model/metrics/system-metrics.yaml

```diff
@@ -105,12 +103,24 @@ groups:
   - id: metric.system.memory.usage
     type: metric
     metric_name: system.memory.usage
-    brief: ""
+    brief: "Reports memory in use by state."
+    note: |
+      The sum over all `system.memory.state` values SHOULD equal the total memory
+      available on the system, that is `system.memory.limit`.
     instrument: updowncounter
     unit: "By"
     attributes:
       - ref: system.memory.state
 
+  - id: metric.system.memory.limit
+    type: metric
+    metric_name: system.memory.limit
+    brief: "Total memory available in the system."
+    note: |
+      Its value SHOULD equal the sum of `system.memory.state` over all states.
+    instrument: updowncounter
```

Comment:
I looked at this now and I'm wondering: Doesn't it make more sense for this metric to be modeled as a Gauge instead? With an UpDownCounter, whomever is recording this would have to do `add(x)` `add(-x)`. Wouldn't it be better to use a Gauge instead, providing a callback and simply get the current available memory? Also, the other metrics also suffer from the same problem like `metric.system.cpu.logical.count`. How an instrumentation would know to "remove" a CPU? 🤔

Reply:
FYI: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#consistent-updowncounter-timeseries

Reply:
I searched for existing `.limit` semantic conventions for metrics. All existing memory related `.limit`s are UpDownCounters, and most but not all `.limit`s are UpDownCounters.

`.limit`s that are UpDownCounters:
- [`jvm.memory.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmmemorylimit)
- [`jvm.buffer.memory.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmbuffermemorylimit)
- [`hw.gpu.memory.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwgpu---gpu-metrics)
- [`hw.logical_disk.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwlogical_disk--logical-disk-metrics)
- [`hw.network.bandwidth.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwnetwork---network-adapter-metrics)
- [`hw.power_supply.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwpower_supply---power-supply-metrics)

`.limit`s that are Gauges:

- [`hw.temperature.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwtemperature---temperature-sensor-metrics)
- [`hw.voltage.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwvoltage---voltage-sensor-metrics)
- [`hw.battery.charge.limit`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/hardware-metrics.md#hwbattery---battery-metrics)

Reply:
For `jvm.memory.limit` the [recommended method to get the value](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--) gives the maximum memory so you would be able to make the same argument wrt using a callback.

Reply:
Thanks for gathering the examples above! I opened https://github.com/open-telemetry/semantic-conventions/issues/438 for us to rethink our current definition for `*.limit`.

Reply:
Thanks! It seems then like UpDownCounter is the right choice for `system.memory.limit`, right? @joaopgrassi do you think we can mark this as resolved?

Reply:
I guess so, yeah.